### PR TITLE
CI: Makefile: Add more flake8 tests and lint the whole repo

### PR DESCRIPTION
**Pull request by cclauss**
## Description ##
(Brief description on what this PR is about)
Makefile: Add more flake8 tests and lint the whole repo
Syntax errors and undefined names in test files are worth knowing about/fixing.

### Essentials ###
- [x] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented

### Changes ###
- [x] more flake8 tests
- [x] lint the whole repo (including tests)

## Comments ##
- Syntax errors and undefined names in test files are worth knowing about/fixing.

### CODE REVIEW

**Comment by cclauss** [comment]
In order to view the Jenkins results I need to provide ci.mxnet.io with read and _write_ access to the code in all my repos, all my settings, all my deployment keys?!?
```
Authorize ci.mxnet.io
@cclauss ci.mxnet.io by dmlc
wants to access your cclauss account
Repositories
Public repositories

This application will be able to read and write all public repository data. This includes the following:

    Code
    Issues
    Pull requests
    Wikis
    Settings
    Webhooks and services
    Deploy keys
```

**Comment by mli** [comment]
Job PR-808/1 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-808/1/index.html

**Comment by cclauss** [comment]
I meant that if I click on the "__Details__" buttons next to the failed tests below, I need to give that access to see the results of those tests.

**Comment by szha** [comment]
@cclauss these are the ones we configured the CI for `read:org,user:email,public_repo`

**Comment by cclauss** [comment]
8241CD3F-B9B8-4A5F-9EE1-7FB554541FBD

**Comment by szha** [comment]
@cclauss we actually don't need most of the settings, but rather just whether a user has admin access to a public repo. would you happen to know what the appropriate scope is?

**Comment by cclauss** [comment]
I do not.  Sorry.

**Comment by szha** [comment]
@cclauss I removed the public-repo part. could you give it another try?

**Comment by cclauss** [comment]
That worked perfectly...  Thanks much.

**Comment by mli** [comment]
Job PR-808/2 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-808/2/index.html

**Comment by mli** [comment]
Job PR-808/3 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-808/3/index.html

File: Makefile [hunk 1]
```diff
@@ -19,7 +19,7 @@ ROOTDIR = $(CURDIR)
 MD2IPYNB = $(ROOTDIR)/docs/md2ipynb.py
 
 flake8:
-	flake8 --exclude conda,*tests*,test_*.py,scripts/word_embeddings/tools/extern --count --select=E901,E999,F821,F822,F823 --show-source --statistics $(lintdir)
+	python3 -m flake8 --exclude conda,scripts/word_embeddings/tools/extern --count --select=E9,F63,F7,F82 --show-source --statistics .
```
Comment (szha) [inline Makefile]: py2 tests don't use python3

Comment (szha) [reply Makefile]: also, how does flake8 help the check for file types other than `.py`?

**Pull request closed without merging**